File: backend-springboot/src/main/java/com/aquamekong/entity/Alert.java
package com.aquamekong.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.OffsetDateTime;

@Entity
@Table(name = "alerts")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class Alert {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "station_id", nullable = false)
    private Station station;

    @Column(name = "alert_type", nullable = false, length = 50)
    private String alertType;

    @Column(nullable = false, length = 20)
    private String severity;       // "WARNING", "CRITICAL", "INFO"

    @Column(nullable = false)
    private String message;

    @Column(name = "threshold_value")
    private Double thresholdValue;

    @Column(name = "actual_value")
    private Double actualValue;

    @Column(name = "is_resolved")
    @Builder.Default
    private Boolean isResolved = false;

    @Column(name = "created_at", nullable = false, updatable = false)
    private OffsetDateTime createdAt;

    @Column(name = "resolved_at")
    private OffsetDateTime resolvedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = OffsetDateTime.now();
    }
}

File: frontend/src/types/alert.ts
export interface AlertDto {
  id: number;
  stationId: number;
  stationName?: string;
  stationCode?: string;
  metricType?: string;
  alertType?: string;
  severity?: 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL' | 'WARNING' | 'INFO';
  alertLevel?: 'CRITICAL' | 'WARNING' | 'SAFE' | 'INFO';
  message?: string;
  thresholdValue?: number;
  measuredValue?: number;
  actualValue?: number;
  isActive?: boolean;
  isResolved?: boolean;
  createdAt: string;
}
